porter/api.py

```
@@ -1,20 +1,44 @@
 """Light wrappers around ``flask`` and ``requests``."""
 
 
+import functools
+import gzip
+import io
 import json
 import uuid
 
 import flask
+import werkzeug.exceptions as werkzeug_exc
+
+from . import config as cf
 
 
 def request_method():
     """Return the HTTP method of the current request, e.g. 'GET', 'POST', etc."""
     return flask.request.method
 
 
-def request_json(*args, **kwargs):
-    """Return the JSON from the current request."""
-    return flask.request.get_json(*args, **kwargs)
+def request_json(silent=False):
+    """Return the JSON from the current request.
+
+    Args:
+        silent (bool): Silence parsing errors and return None instead.
+    """
+    request = flask.request
+    encoding = request.content_encoding
+    try:
+        if encoding == 'gzip':
+            return json.loads(gzip.decompress(request.get_data()).decode('utf-8'))
+        elif encoding in ('identity', None):
+            return request.get_json(force=True)
+    except Exception as err:
+        if not silent:
```

What's the use case for ignoring an error here?

---

It's used by `response.make_error_response()` if `return_user_data_on_error`.